libs/langchain/langchain/agents/tool_calling_agent/base.py

```diff
@@ -6,13 +6,17 @@
 from langchain_core.tools import BaseTool
 
 from langchain.agents.format_scratchpad.tools import (
+    ToolMessagesFormatter,
     format_to_tool_messages,
 )
 from langchain.agents.output_parsers.tools import ToolsAgentOutputParser
 
 
 def create_tool_calling_agent(
-    llm: BaseLanguageModel, tools: Sequence[BaseTool], prompt: ChatPromptTemplate
+    llm: BaseLanguageModel,
+    tools: Sequence[BaseTool],
+    prompt: ChatPromptTemplate,
+    tool_messages_formatter: ToolMessagesFormatter = format_to_tool_messages,
```

Comment:
We need a better name for this argument.

Here are some options:

* `message_formatter`
* `history_to_message_formatter`
* `agent_history_to_messages`